lib/stripe/stripe_client.rb

```diff
@@ -382,12 +414,15 @@ def self.maybe_gc_connection_managers
 
       pruned_thread_contexts = []
       @thread_contexts_with_connection_managers.each do |thread_context|
-        connection_manager = thread_context.default_connection_manager
-        next if connection_manager.last_used > last_used_threshold
-
-        connection_manager.clear
-        thread_context.default_connection_manager = nil
-        pruned_thread_contexts << thread_context
+        thread_context
+          .default_connection_managers
+          .each do |_, connection_manager|
+            next if connection_manager.last_used > last_used_threshold
+
+            connection_manager.clear
+            thread_context.reset_connection_managers
```

Comment:
Shouldn't we only remove the connection manager that we're currently looping over rather than all the ones on the thread?

Reply:
Good call! I'll update this to clear the single connection manager.